### PR TITLE
cm2kc: clustermap file to kubeconfig file converter

### PR DESCRIPTION
**Pull request by clarketm**
`cm2kc` is a CLI tool used to convert a **clustermap file** to a **kubeconfig file**.

fix https://github.com/kubernetes/test-infra/issues/16381

> Note: `input` and `output` can be piped to/from `kubectl` (or other) commands.

> FYI: unless there is utility in preservation, this tool will be deleted along with https://github.com/kubernetes/test-infra/pull/14724 in **May 2020**.

### CODE REVIEW

File: prow/cmd/cm2kc/main_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+			os.Args = append(os.Args, "--input="+in, "--output="+outA)
+			main()
+
+			actual, err := ioutil.ReadFile(outA)
+			if err != nil {
+				t.Errorf("Failed reading actual output file %v: %v", outA, err)
+			}
+
+			if os.Getenv("VERBOSE") == "true" {
+				fmt.Printf("expected (%v):\n%v\n", test.name, string(expected))
```
Comment (cjwagner) [inline prow/cmd/cm2kc/main_test.go]: nit: consider `t.Logf`

**Comment by clarketm** [comment]
/unhold

New example added. PTAL

File: prow/cmd/cm2kc/main_test.go (hunk 1 above)
Comment (clarketm) [reply prow/cmd/cm2kc/main_test.go]: Updated. Thanks for the info!

File: prow/cmd/cm2kc/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+#### Add a kubeconfig file in a secret: `kubeconfig` from a clustermap file in another secret: `build-cluster` for context: `my-context`.
+
+The following command will:
+1. Get a *clustermap* formatted secret: `build-cluster` in key: `cluster` for context: `my-context`.
+1. Base64 decode the secret.
+1. Convert the *clustermap* data to a *kubeconfig* format.
+1. Create a *kubeconfig* formatted secret: `kubeconfig` in key: `config` for context: `my-context` from the converted data.
+
+```shell
+kubectl --context=my-context secrets build-cluster -o jsonpath='{.data.cluster}' |
```
Comment (cjwagner) [inline prow/cmd/cm2kc/README.md]: kubectl **get**

File: prow/cmd/cm2kc/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  base64 -d |
+  bazel run //prow/cmd/cm2kc |
+  kubectl --context=my-context secret generic kubeconfig --from-file=config=/dev/stdin
```
Comment (cjwagner) [inline prow/cmd/cm2kc/README.md]: kubectl **create**

File: prow/cmd/cm2kc/README.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+```
```
Comment (cjwagner) [inline prow/cmd/cm2kc/README.md]: Also indicate that the final step after migrating the secret is to switch Prow components that specify `--build-cluster` to specify `--kubeconfig` instead.

File: prow/cmd/cm2kc/README.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  user:
+    client-certificate-data: fake-default-ca-cert
+    client-key-data: fake-default-ca-cert
+- name: build
+  user:
+    client-certificate-data: fake-build-ca-cert
+    client-key-data: fake-build-ca-cert
+```
+
+#### Add a kubeconfig file in a secret: `kubeconfig` from a clustermap file in another secret: `build-cluster` for context: `my-context`.
```
Comment (cjwagner) [inline prow/cmd/cm2kc/README.md]: nit: I'd put this example first and also indicate that it is probably the only thing a Prow admin needs to look at to do the migration.

File: prow/cmd/cm2kc/README.md (hunk 2 above)
Comment (clarketm) [reply prow/cmd/cm2kc/README.md]: Oops, cannot explain that paste error 🤦‍♂.

**Pull request merged**